rdrf/report/report_builder.py

```diff
@@ -67,8 +73,18 @@ def __get_graphql_query(self, offset=None, limit=None):
 
         fields_nested_demographics = []
         for model_name, fields in other_demographic_fields.items():
-            fields_demographic = GqlQuery().fields(fields).query(model_name).generate()
-            fields_nested_demographics.append(fields_demographic)
+            model_config = self.report_config[model_name]
+            if model_config.get('pivot', False):
+                # Lookup the variants of this item which will form the column header groupings
+                # e.g. for consents, returns a list of the unique consent codes
+                column_headers = self.__get_variants(model_config.get('variant_lookup'))
+
+                # For each grouping, generate the query containing each of the fields selected
+                col_queries = [GqlQuery().fields(fields).query(header).generate() for header in column_headers]
+                fields_nested_demographics.append(GqlQuery().fields(col_queries).query(model_name).generate())
```

Comment:
If I have a report that queries consent answers, but there aren't any consent questions, the gql query this generates causes a syntax error because the inner `consents` values are empty. The same might apply to other nested demographics?

```
Syntax Error GraphQL (1:142) Expected Name, found }\n\n1: query { patients(registryCode: "ang", consentQuestionCodes: [], workingGroupIds: [], offset: 1, limit: 1) { familyName givenNames consents { } } }\n                                                                                                                                                ^\n'
```

Reply:
Maybe fix with a PR to [youyo/gql-query-builder](https://github.com/youyo/gql-query-builder)?

Reply:
It's probably more correct for me to check my variable isn't null before I try to generate the query, rather than silently handle an exceptional condition (by modifying the builder to remove the empty node).

Reply:
Yeah, you can check as well. But that library shouldn't be generating queries with syntax errors; it should know when to drop certain fields, or throw an exception when given invalid input.

Reply:
I'll look into adding a PR for the changes to the query builder.